ydb/query/pool.py

```diff
@@ -22,20 +27,71 @@
 class QuerySessionPool:
     """QuerySessionPool is an object to simplify operations with sessions of Query Service."""
 
-    def __init__(self, driver: common_utils.SupportedDriverType):
+    def __init__(self, driver: common_utils.SupportedDriverType, size: int = 100):
         """
         :param driver: A driver instance
         """
 
         logger.warning("QuerySessionPool is an experimental API, which could be changed.")
         self._driver = driver
-
-    def checkout(self) -> "SimpleQuerySessionCheckout":
+        self._queue = queue.PriorityQueue()
+        self._current_size = 0
+        self._size = size
+        self._should_stop = threading.Event()
+        self._lock = threading.RLock()
+
+    def _create_new_session(self, timeout: float):
+        session = QuerySessionSync(self._driver)
+        session.create(settings=BaseRequestSettings().with_timeout(timeout))
+        logger.debug(f"New session was created for pool. Session id: {session._state.session_id}")
+        return session
+
+    def acquire(self, timeout: float) -> QuerySessionSync:
+        with self._lock:
+            if self._should_stop.is_set():
+                logger.error("An attempt to take session from closed session pool.")
+                raise RuntimeError("An attempt to take session from closed session pool.")
+
+            session = None
+            try:
+                _, session = self._queue.get_nowait()
+            except queue.Empty:
+                pass
+
+            start = time.monotonic()
+            if session is None and self._current_size == self._size:
+                try:
+                    _, session = self._queue.get(block=True, timeout=timeout)
```

Comment:
it is long operation under mutex lock.

How you will enforce smaller timeout on parallel request?